src/modules/px4iofirmware/mixer.cpp

```diff
@@ -199,39 +200,41 @@ mixer_tick(void)
 	 *
 	 */
 	should_arm = (
-			     /* IO initialised without error */ (r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)
-			     /* and IO is armed */ 		  && (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_OFF)
-			     /* and FMU is armed */ 		  && (
-				     ((r_setup_arming & PX4IO_P_SETUP_ARMING_FMU_ARMED)
-				      /* and there is valid input via or mixer */         && (r_status_flags & PX4IO_P_STATUS_FLAGS_MIXER_OK))
-				     /* or direct PWM is set */               || (r_status_flags & PX4IO_P_STATUS_FLAGS_RAW_PWM)
-				     /* or failsafe was set manually */	 || ((r_setup_arming & PX4IO_P_SETUP_ARMING_FAILSAFE_CUSTOM)
-						     && !(r_status_flags & PX4IO_P_STATUS_FLAGS_FMU_OK))
+			     (r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)				/* IO initialised without error */
+			     && (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_OFF)			/* and IO is armed */
+			     && (
+				     ((r_setup_arming & PX4IO_P_SETUP_ARMING_FMU_ARMED)		/* and FMU is armed */
+				      && (r_status_flags & PX4IO_P_STATUS_FLAGS_MIXER_OK))	/* and there is valid input via or mixer */
+				     || (r_status_flags & PX4IO_P_STATUS_FLAGS_RAW_PWM)		/* or direct PWM is set */
 			     )
 		     );
 
 	should_arm_nothrottle = (
-					/* IO initialised without error */ (r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)
-					/* and IO is armed */ 		  && (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_OFF)
-					/* and there is valid input via or mixer */         && (r_status_flags & PX4IO_P_STATUS_FLAGS_MIXER_OK));
+					(r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)		/* IO initialised without error */
+					&& (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_OFF)	/* and IO is armed */
+					&& (r_status_flags & PX4IO_P_STATUS_FLAGS_MIXER_OK)	/* and there is valid input via or mixer */
+				);
 
-	should_always_enable_pwm = (r_setup_arming & PX4IO_P_SETUP_ARMING_ALWAYS_PWM_ENABLE)
-				   && (r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)
-				   && (r_status_flags & PX4IO_P_STATUS_FLAGS_FMU_OK);
+	should_always_enable_pwm = (
+					   (r_setup_arming & PX4IO_P_SETUP_ARMING_ALWAYS_PWM_ENABLE)
+					   && (r_status_flags & PX4IO_P_STATUS_FLAGS_INIT_OK)
+					   && (r_status_flags & PX4IO_P_STATUS_FLAGS_FMU_OK)
+				   );
 
 	/*
 	 * Check if failsafe termination is set - if yes,
 	 * set the force failsafe flag once entering the first
 	 * failsafe condition.
 	 */
-	if ( /* if we have requested flight termination style failsafe (noreturn) */
-		(r_setup_arming & PX4IO_P_SETUP_ARMING_TERMINATION_FAILSAFE) &&
-		/* and we ended up in a failsafe condition */
-		(source == MIX_FAILSAFE) &&
+	if ( /* Flight termination is allowed */
+		REG_TO_BOOL(r_setup_flighttermination) &&
```

Comment:
The indenting is off here, I think. I would prefer the comments to  be on the right like above.